src/dfx-core/src/network/root_key.rs

```diff
@@ -0,0 +1,34 @@
+use ic_agent::Agent;
+
+use crate::{
+    config::model::network_descriptor::NetworkDescriptor, error::root_key::FetchRootKeyError,
+};
+
+pub async fn fetch_root_key_if_needed(
+    agent: &Agent,
+    network: &NetworkDescriptor,
+) -> Result<(), FetchRootKeyError> {
+    if !network.is_ic {
+        agent
+            .fetch_root_key()
+            .await
+            .map_err(FetchRootKeyError::ReplicaError)?;
+    }
+    Ok(())
+}
+
+/// Fetches the root key of the local network.
+/// Returns an error if attempted to run on the real IC.
+pub async fn fetch_root_key_or_anyhow(
```

Comment:
We need a better name here.  `_or_anyhow` is referring to `anyhow!()`.  In context this means DfxResult, and note that dfx-core does not use `anyhow`.

No good suggestions for a name, but here is one: `fetch_root_key_but_fail_if_ic`